docs/auto-publish-products-to-sales-channels-by-tag/script.liquid

```diff
@@ -0,0 +1,182 @@
+{% assign sales_channel_names = options.sales_channel_names__array_required %}
+{% assign product_tag = options.product_tag__required %}
+
+{% capture query %}
+  query {
+    publications(first: 250) {
+      edges {
+        node {
+          id
+          name
+        }
+      }
+    }
+  }
+{% endcapture %}
+
+{% assign result = query | shopify %}
+
+{% if event.preview %}
+  {% capture result_json %}
+    {
+      "data": {
+        "publications": {
+          "edges": [
+            {
+              "node": {
+                "id": "gid://shopify/Publication/1234567890",
+                "name": "Online Store"
+              }
+            },
+            {
+              "node": {
+                "id": "gid://shopify/Publication/2345678901",
+                "name": "Buy Button"
+              }
+            }
+          ]
+        }
+      }
+    }
+  {% endcapture %}
+
+  {% assign result = result_json | parse_json %}
+{% endif %}
+
+{% assign publications = result.data.publications.edges | map: "node" %}
+{% assign publications_indexed_by_name = publications | index_by: "name" %}
```

Comment:
Log this value, for debugging: `{% log available_publications: publications_indexed_by_name %}`, maybe. Small mistakes on sales channel names are common, and logging all the available ones helps when ironing out those config issues.